ghc-lib-gen.cabal

```diff
@@ -52,7 +52,10 @@ executable ghc-lib-gen
 
 executable ghc-lib-build-tool
   import: base
-  build-tool-depends: alex:alex, happy:happy < 2.0
+  if impl (ghc > 9.12.0)
```

Comment:
Should this bound instead check for 9.8?

Reply:
no, it's a bit subtle. this choice affects the happy that we'll use in ghc-lib-gen that invokes the GHC configure stage in order to compute the modules that go into each of ghc-lib-parser and ghc-lib, not the happy build-tool-depends of those packages (building these packages does not involve configure and generally don't fail with happy-2.0.2, i.e. the point raised in #563 ). except on the current HEAD of the GHC source tree, configure will fail with happy > 1.20 (the change that admits happy 2.0.2 only went on master 2 weeks ago {see the referenced commit above}). so this checks that the GHC source we are working with is at least > ghc-9.12.1 (equivalently, the minimum ghc-lib-parser flavor is ghc-9.12.1) before enabling 2.0.2 for happy for the purposes of ghc-lib-gen. in all other cases, 1.20.* works and will pass configure just fine so in tl;dr, this is what we want.